<!DOCTYPE html>
<html>
<head>
<link rel="stylesheet" href="styling2.css">
</head>
<body>



<h1>HELP</h1>

<script src = "styling2.js" defer></script>

<button class = "button", id = "jobSetup", onclick = "function()"> Job Setup</button>
<div id="btn-group1" hidden>
  <h2>NanoString data</h2>
  <p>Your NanoString data should be loaded as a ZIP folder containing one 
  RCC file per sample, or a CSV or TXT file containing expression data 
  for all samples.</p>
  
  <h2>Sample info table</h2>
  <p>This takes a CSV file containing sample information.  This Shiny app conducts
  group-vs.-group comparisons, so one column must include the group identifier for each sample (for example, Treated or Control). 
  Sample names should be in the first column. If the RCC filenames are used as sample
  names, they will be merged so that they match.  Otherwise, the RCC files and sample
  table will be merged in order. The merge can be checked prior to analysis using "Check Samples", to make
  sure they match.</p>
  
  <h2>Group Column</h2>
  <p>After loading the sample info table, a dropdown box will appear, containing the columns
  from that table (such as the "Treatment" column). Select the column containing the group 
  information for differential expression analysis.</p>
  
  <h2>Base Group</h2>
  <p>This is the group that all other groups will be compared against (i.e. the 
  denominator in your Fold Change). For example, if your data contain groups "A", "B", 
  and "C", and you select "C" as the base group, NanoTube will conduct analyses and 
  calculate fold changes for A vs. C and B vs. C.</p>
  
  <h2>Gene Set Database</h2>
  <p>Required only for Gene Set Analysis feature. Gene set database (gmt format, such as those found at MSigDB)
  can beloaded here.</p>
  
  <h2>Advanced Options</h2>
  <h3>Housekeeping Genes</h3>
  <p>NanoTube will automatically identify housekeeping genes for normalization by selecting
  those marked as "Housekeeping" in the <b>CodeClass</b>. Alternatively, these can be manually
  specified with this option, by inputting a comma-separated list of genes to be used as housekeeping genes.</p>
  
  <h3>Negative Control Threshold</h3>
  <p>The NanoTube app performs a two-sample, one-sided t test for each gene: this test compares the
  expression of that gene across all samples vs. the expression of all negative control genes
  across all samples. The p-value of this test is for the null hypothesis that the endogenous gene does
  not have expression above the negative control genes. In this field, you can specify the signifance
  level required to keep a gene for analysis. A lower value is a more stringent threshold: 
  genes with a p-value above this level will be removed, and not
  included in differential expression or gene set analysis.</p>
  
  <h3>Minimum Gene Set Size</h3>
  <p>In this field, you can set the number of genes required for an individual gene set (from a loaded gmt file)
  to be included in analysis. For example, if this value is set to 5 (default), only gene sets with at least 5 genes present
  in your NanoString data set will be included in gene set analysis.</p>
  
</div>
<br>

<button class = "button", id = "QCResults", onclick = "function()">QC Results</button>
<div id="btn-group2" hidden>
  <h2>Positive Controls</h2>
  <p>The NanoTube web app identifies positive control, negative control, and housekeeping genes prior to analysis,
  mainly based on each gene's <b>CodeClass</b>.  Positive control normalization is performed by scaling all genes by 
  the <b>Scale Factor</b> for each sample, calculated using the geometric mean of the positive control genes. 
  Scale factors outside of the range 0.3-3 indicate high variance between samples and potential problems in the 
  data, and these should be checked before proceeding.</p>
  
  <p>A scatter plot of Observed vs. Expected expression of positive control genes is also provided
  for each sample, with the <b>R-squared</b> values provided in the table. "Expected" expression is
  identified from the positive control gene names.  For example, the gene POS_I(8) has an expected expression
  of 8 (or expected log2 expression of 2). Low R-squared values potentially indicate issues in detection of
  the positive control genes.</p>
  
  <h2>Negative Controls</h2>
  <p>A table showing statistics for the detection of negative control genes is provided first, including the average,
  maximum, and standard deviation of negative control gene detection. In addition, a threshold "Background" level is 
  provided, and calculated as mean + 2*sd. The number (and percentage) of endogenous genes with average 
  expression below that level is presented for each sample. Samples with a high percentage of genes detected below
  background level may present a quality concern.</p>
  
  <p>A scatter chart of the detected counts of each negative control in each sample is also provided.</p>

  
  <h2>Housekeeping</h2>
  <p>As with positive control genes, a scale factor is calculated using the geometric mean of the 
  housekeeping genes for each sample, and this is used to normalize the endogenous genes. housekeeping
  genes are automatically detected from RCC files, or they can be manually entered in the Advanced Options
  on the "Job Setup" tab. Scale factors outside the range of 0.3-3 indicate high variance between samples, and
  should be checked before proceeding.</p>
  
  <p>Boxplots of endogenous gene expression before and after normalization are also provided in this tab.</p>
</div>
<br>

<button class = "button", id = "analyzeResults", onclick = "function()">Analyze Results</button>
<div id="btn-group3" hidden>
  
  <h2>PCA</h2>
  <p>A simple Principle Components Analysis is performed using the 'prcomp' function, and 
  a scatterplot of the first two principle components is drawn here. Samples are represented as points,
  colored by their Group membership, and specific sample names can be identified by hovering over the points.</p>
  
  <h2>Volcano Plot</h2>
  <p>A volcano plot of the selected group vs. the base group is drawn here.  Fold change is calculated as FC =
  (Selected Group / Base Group), so genes on the right of the plot have higher expression in the Selected Group. 
  A log2(FC) and nominal p-value cutoff can be defined in the boxes, to draw vertical and horizontal lines (respectively)
  on the plot. Target names (such as gene symbols) can be identified by hovering over the points.</p>
  
  <h2>Differential Expression</h2>
  <p>For each analysis of Group vs. Base Group (indicated in parentheses), a table containing the following four
  columns is provided:</p>
  <p><b>Log2FC</b>: The log2 fold change of each gene, calculated as log2(Group / Base Group). Genes with positive 
  log2FC have higher expression in "Group", while those with negative log2FC have higher expression in "Base Group".</p>
  <p><b>t</b>: The comparison statistic calculated by 'limma'.</p>
  <p><b>p-val</b>: The nominal p-value calculated by 'limma'.</p>
  <p><b>q-val</b>: The p-value adjusted for multiple testing, using the Benjamini-Hochberg procedure.</p>
  
</div>
<br>




<button class = "button", id = "geneSetAnalysis", onclick = "function()">Gene Set Analysis</button>
<div id="btn-group4" hidden>
  
  <p>Gene set enrichment analysis is conducted using the 'fgsea' preranked package,
  resulting in a table of statistics and a heatmap of gene expression within specified
  gene sets.</p>
  
  <h2>Options</h2>
  <p><b>Comparison</b>: The group that will be compared against the base group.</p>
  <p><b>Clustering Threshold</b>: Gene sets containing similar genes are clustered together to ease the 
  interpretation of results. The overlap between two gene sets is calculated as the Jaccard index,
  the genes overlapping between the sets divided by the union (total genes) of the sets.  Sets with an overlap
  greater than the clustering threshold will be "clustered" together.</p>
  <p><b>q-value Threshold</b>: Only gene sets with a q-value (p-value adjusted for multiple testing) 
  below the threshold will be included in the table or heatmap. Use a threshold of "1" to 
  include all gene sets.</p>
  <p><b>Cluster to plot</b>: The gene set "cluster" to plot in a heatmap. The assigned
  cluster for each gene set can be found in the results table; a cluster may contain
  one or more gene sets.</p>
  
  <h2>Result Table</h2>
  <p>This table contains the standard gene set analysis statistics for each gene set. "p.adj" is equivalent
  to a "q-value", and the NES provides the normalized enrichment score, or the direction of the enrichment.
  Gene sets that are more highly expressed in the "Group" will have a positive NES, while those highly expressed
  in "Base Group" will have a negative NES.</p>
  
  <p>Gene sets are clustered using the Jaccard index (as defined above), and the gene set with the lowest p-value
  within each cluster is defined as the "most enriched" gene set of the cluster ("Cluster.Max").</p>
  
  <h2>Heatmap</h2>
  <p>A leading edge heatmap for a "Cluster" of gene sets can be plotted by specifying which cluster in the options.
  Cells are colored based on the log2(expression) for each gene in each sample, centered on the median of the 
  "Base Group". If there are more than one gene set in the cluster, gene set membership information is also
  provided in the heatmap.</p>
  
  
</div>
<br>









</body>
</html>